src/Compilers/Core/MSBuildTask/ManagedCompiler.cs

```diff
@@ -305,7 +305,7 @@ protected override Encoding StandardOutputEncoding
 
         protected override int ExecuteTool(string pathToTool, string responseFileCommands, string commandLineCommands)
         {
-            if (!UseSharedCompilation || this.ToolPath != null )
+            if (!UseSharedCompilation || !String.IsNullOrEmpty(this.ToolPath))
```

Comment:
I'm concerned about this.  Say I want to set the build to use `csc / vbc` which is the current directory.  Today I could do this by doing the following:

``` xml
<CscToolPath><CscToolPath>
```

After this change that would cease to work.  It could still be done with something like `$(MSBuildFileCurrentDirectory)` probably.  My instinct though is an empty `CscToolPath` should be treated as a path.